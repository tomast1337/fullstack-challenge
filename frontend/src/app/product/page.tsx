import { ProductSearchList } from '@frontend/components/product/ProductSearchList';

export default async function Page() {
  return <ProductSearchList />;
}
