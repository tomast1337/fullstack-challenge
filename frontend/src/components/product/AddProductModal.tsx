'use client';
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
} from '@frontend/components/ui/dialog';
import { useToast } from '@frontend/hooks/use-toast';
import { ProductForm, ProductSchemaType } from './ProductSchemaType';

export const AddProductModal = ({
  isOpen,
  onClose,
  onAdd,
  product = {
    name: '',
    price: 0,
    category: '',
    description: '',
    stockQuantity: 0,
  },
}: {
  isOpen: boolean;
  onClose: () => void;
  onAdd: (data: ProductSchemaType) => void;
  product?: ProductSchemaType;
}) => {
  const { toast } = useToast();

  const handleAdd = async (data: ProductSchemaType) => {
    try {
      await onAdd(data);
      onClose();
      toast({
        title: 'Product added',
        description: 'Product has been added successfully',
      });
    } catch {
      toast({
        title: 'Error',
        description: 'An error occurred. Please try again',
      });
    }
  };

  return (
    <Dialog open={isOpen} onOpenChange={onClose}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Add Product</DialogTitle>
          <DialogDescription>
            Fill in the form below to add a new product
          </DialogDescription>
        </DialogHeader>
        <ProductForm product={product} onSubmit={handleAdd} />
      </DialogContent>
    </Dialog>
  );
};
