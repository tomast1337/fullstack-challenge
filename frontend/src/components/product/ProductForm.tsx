'use client';
import { Input } from '@frontend/components/ui/input';
import { zodResolver } from '@hookform/resolvers/zod';
import { Label } from '@radix-ui/react-label';
import { useEffect } from 'react';
import { useForm } from 'react-hook-form';
import * as zod from 'zod';
import { Button } from '../ui/button';

const validationSchema = zod.object({
  name: zod.string().nonempty('Name is required'),
  price: zod.number().positive('Price must be a positive number'),
  category: zod.string().nonempty('Category is required'),
  description: zod.string().nonempty('Description is required'),
  stockQuantity: zod
    .number()
    .positive('Stock quantity must be a positive number'),
  id: zod.string().optional(),
  image: zod.string().optional(),
});
export type ProductSchemaType = zod.infer<typeof validationSchema>;
export const ProductForm = ({
  product,
  onSubmit,
}: {
  product?: ProductSchemaType;
  onSubmit: (data: ProductSchemaType) => void;
}) => {
  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors },
  } = useForm<ProductSchemaType>({
    resolver: zodResolver(validationSchema),
  });

  useEffect(() => {
    if (product) {
      Object.keys(product).forEach((key) => {
        setValue(
          key as keyof ProductSchemaType,
          product[key as keyof ProductSchemaType],
        );
      });
    }
  }, [product, setValue]);

  return (
    <form onSubmit={handleSubmit(onSubmit)} className='flex flex-col gap-4'>
      <Label htmlFor='name' className='text-sm text-gray-300'>
        Name
      </Label>
      <Input
        id='name'
        placeholder='Enter product name'
        className='rounded-lg'
        {...register('name')}
      />
      {errors.name && <p className='text-red-500'>{errors.name.message}</p>}
      <Label htmlFor='price' className='text-sm text-gray-300'>
        Price
      </Label>
      <Input
        id='price'
        placeholder='Enter product price'
        className='rounded-lg'
        {...register('price')}
      />
      {errors.price && <p className='text-red-500'>{errors.price.message}</p>}
      <Label htmlFor='category' className='text-sm text-gray-300'>
        Category
      </Label>
      <Input
        id='category'
        placeholder='Enter product category'
        className='rounded-lg'
        {...register('category')}
      />
      {errors.category && (
        <p className='text-red-500'>{errors.category.message}</p>
      )}
      <Label htmlFor='description' className='text-sm text-gray-300'>
        Description
      </Label>
      <Input
        id='description'
        placeholder='Enter product description'
        className='rounded-lg'
        {...register('description')}
      />
      {errors.description && (
        <p className='text-red-500'>{errors.description.message}</p>
      )}
      <Label htmlFor='stockQuantity' className='text-sm text-gray-300'>
        Stock Quantity
      </Label>
      <Input
        id='stockQuantity'
        placeholder='Enter product stock quantity'
        className='rounded-lg'
        {...register('stockQuantity')}
      />
      {errors.stockQuantity && (
        <p className='text-red-500'>{errors.stockQuantity.message}</p>
      )}
      <Button
        type='submit'
        className='bg-blue-600 hover:bg-blue-500 transition rounded-lg flex items-center gap-2'
      >
        Save
      </Button>
    </form>
  );
};
