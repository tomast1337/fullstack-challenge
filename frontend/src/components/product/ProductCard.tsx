'use client';
import { ProductDto } from '@backend/product/dto/product.dto';
import Image from 'next/image';
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from '@frontend/components/ui/card';
import { Skeleton } from '@frontend/components/ui/skeleton';
import InfoIcon from '@mui/icons-material/Info';
import { Button } from '@frontend/components/ui/button';
import { useRouter } from 'next/navigation';
export const ProductCardSkeleton = () => (
  <Card className='w-[300px] max-w-sm'>
    {/* Skeleton for Product Image */}
    <Skeleton className='w-full h-48 rounded-t-lg' />

    {/* Skeleton for Card Header (Product Name and Description) */}
    <CardHeader>
      <CardTitle>
        <Skeleton className='w-3/4 h-6' />
      </CardTitle>
      <CardDescription>
        <Skeleton className='w-full h-4 mt-2' />
        <Skeleton className='w-2/3 h-4 mt-2' />
      </CardDescription>
    </CardHeader>

    {/* Skeleton for Card Content (Product Price) */}
    <CardContent>
      <Skeleton className='w-1/4 h-6' />
    </CardContent>

    {/* Skeleton for Card Footer (Add to Cart Button) */}
    <CardFooter>
      <Skeleton className='w-full h-10 rounded-lg' />
    </CardFooter>
  </Card>
);
export const LoadingProducts = ({ length }: { length: number }) => {
  return (
    <>
      {[...Array(length)].map((_, index) => (
        <ProductCardSkeleton key={index} />
      ))}
    </>
  );
};
export const ProductCard = ({ product }: { product: ProductDto }) => {
  const { description, name, picture, price } = product;
  const router = useRouter();

  const onProductClick = () => {
    router.push(`/product/${product.id}`);
  };
  return (
    <Card className='w-[300px] max-w-sm h-[500px] flex flex-col'>
      {/* Card Header: Product Image and Name */}
      <CardHeader>
        <Image
          src={picture}
          alt={name}
          width={300}
          height={200}
          className='w-full h-48 object-cover rounded-t-lg'
        />
        <CardTitle className='mt-4'>{name}</CardTitle>
        <CardDescription>{description}</CardDescription>
      </CardHeader>

      {/* Card Content: Product Price */}
      <CardContent>
        <p className='text-lg font-semibold'>${price.toFixed(2)}</p>
      </CardContent>

      <span className='flex-grow' />
      {/* Card Footer: See Product Button */}
      <CardFooter>
        {product.stockQuantity > 0 ? (
          <Button
            onClick={onProductClick}
            className='w-full bg-blue-500 text-white py-2 px-4 rounded-lg hover:bg-blue-600 transition-colors'
          >
            <InfoIcon className='mr-2' />
            See Product
          </Button>
        ) : (
          <Button
            className='w-full bg-red-500 text-white py-2 px-4 rounded-lg hover:bg-red-600 transition-colors'
            disabled
          >
            Out of Stock
          </Button>
        )}
      </CardFooter>
    </Card>
  );
};
