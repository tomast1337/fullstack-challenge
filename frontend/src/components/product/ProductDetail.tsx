'use client';
import AddShoppingCartIcon from '@mui/icons-material/AddShoppingCart';
import { ProductDto } from '@backend/product/dto/product.dto';
import { Button } from '@frontend/components/ui/button';
import Image from 'next/image';
import { ProductSugesstion } from './ProductSugesstion';
import { useOrder } from '@frontend/hooks/use-order';
import { useShallow } from 'zustand/react/shallow';
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from '../ui/select';
import { useState } from 'react';
import { useToast } from '@frontend/hooks/use-toast';
import { Label } from '../ui/label';

export const ProductDetail = ({ product }: { product: ProductDto }) => {
  const { addToOrder } = useOrder(useShallow((state) => ({ ...state })));
  const { toast } = useToast();
  const [quantity, setQuantity] = useState(1);

  const handleAddToCart = async () => {
    if (product.stockQuantity < quantity) {
      toast({
        title: 'Error',
        description: 'Not enough stock',
      });
      return;
    }
    try {
      await addToOrder(product.id, quantity);
      toast({
        title: 'Success',
        description: 'Product added to cart',
      });
    } catch {
      toast({
        title: 'Error',
        description: 'An error occurred. Please try again',
      });
    }
  };

  return (
    <div className='w-[90%] container mx-auto px-auto py-8 flex flex-col space-y-8 h-full'>
      <div className='grid grid-cols-1 md:grid-cols-2 gap-8'>
        {/* Product Image */}
        <div>
          <Image
            width={500}
            height={500}
            src={product.picture}
            alt={product.name}
            className='max-w-80 max-h-80 object-cover rounded-t-lg'
          />
        </div>

        {/* Product Details */}
        <div className='space-y-4'>
          <h1 className='text-3xl font-bold'>{product.name}</h1>
          <p className='text-gray-600'>{product.description}</p>
          <p className='text-2xl font-semibold'>${product.price.toFixed(2)}</p>
          {product.stockQuantity > 0 ? (
            <>
              <Label>Set the quantity:</Label>
              <Select>
                {/* Select Trigger */}
                <SelectTrigger className='w-[180px]'>
                  <SelectValue placeholder='1' />
                </SelectTrigger>
                <SelectContent>
                  {/* Select Items */}
                  {Array.from({ length: product.stockQuantity }).map(
                    (_, index) => (
                      <SelectItem
                        onClick={() => setQuantity(index + 1)}
                        key={index}
                        value={String(index + 1)}
                      >
                        {index + 1}
                      </SelectItem>
                    ),
                  )}
                </SelectContent>
              </Select>

              {/* Add to Cart Button */}
              <Button
                className='w-full bg-blue-500 text-white py-2 px-4 rounded-lg hover:bg-blue-600 transition-colors'
                onClick={handleAddToCart}
              >
                <AddShoppingCartIcon className='mr-2' />
                Add to Cart
              </Button>
            </>
          ) : (
            <p className='text-red-500'>Out of Stock</p>
          )}
        </div>
      </div>
      <span className='flex-grow' />
      <div className='flex flex-col flex-grow align-middle space-y-4 w-[90%] mx-auto'>
        <ProductSugesstion />
      </div>
    </div>
  );
};
