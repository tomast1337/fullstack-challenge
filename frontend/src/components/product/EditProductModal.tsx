'use client';
import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogHeader,
  DialogTitle,
} from '@frontend/components/ui/dialog';
import { useToast } from '@frontend/hooks/use-toast';
import { ProductForm, ProductSchemaType } from './ProductForm';

export const EditProductModal = ({
  onEdit,
  product = {
    name: '',
    price: 0,
    category: '',
    description: '',
    stockQuantity: 0,
  },
  isOpen,
  onClose,
}: {
  onEdit: (data: ProductSchemaType) => void;
  product?: ProductSchemaType;
  isOpen: boolean;
  onClose: () => void;
}) => {
  const { toast } = useToast();

  const handleEdit = async (data: ProductSchemaType) => {
    try {
      await onEdit(data);
      toast({
        title: 'Product updated',
        description: 'Product has been updated successfully',
      });
    } catch {
      toast({
        title: 'Error',
        description: 'An error occurred. Please try again',
      });
    }
  };

  return (
    <Dialog open={isOpen} onOpenChange={onClose}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle>Editing Product {product.name}</DialogTitle>
          <DialogDescription>
            Fill in the form below to edit the product
          </DialogDescription>
        </DialogHeader>
        <ProductForm product={product} onSubmit={handleEdit} />
      </DialogContent>
    </Dialog>
  );
};
