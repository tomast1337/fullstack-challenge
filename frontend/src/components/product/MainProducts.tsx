'use client';
import { UseProducts, useProducts } from '@frontend/hooks/use-products';
import { useEffect } from 'react';
import { useShallow } from 'zustand/react/shallow';
import { LoadingProducts, ProductCard } from './ProductCard';
import { ProductPagination } from './ProductPagination';

export const MainProducts = () => {
  const {
    fetchProducts,
    products,
    loading,
    total,
    page,
    limit,
    query,
  }: UseProducts = useProducts(useShallow((state) => state));

  const pageSize = 25;

  useEffect(() => {
    fetchProducts(1, pageSize, '');
  }, [page, limit, query, fetchProducts]);

  return (
    <section className='flex flex-wrap gap-4 justify-center'>
      {loading ? (
        <LoadingProducts length={pageSize} />
      ) : (
        <>
          {products?.map((product) => (
            <ProductCard key={product.id} product={product} />
          ))}
        </>
      )}
      {/* Pagination Component */}
      {total > pageSize && (
        <ProductPagination
          current={page}
          limit={pageSize}
          total={total}
          query=''
        />
      )}
    </section>
  );
};
