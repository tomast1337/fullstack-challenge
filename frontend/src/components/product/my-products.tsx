'use client';

import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from '@frontend/components/ui/table';
import { useProductStore } from '@frontend/hooks/use-product-store';
import AddIcon from '@mui/icons-material/Add';
import DeleteIcon from '@mui/icons-material/Delete';
import EditIcon from '@mui/icons-material/Edit';
import VisibilityIcon from '@mui/icons-material/Visibility';
import { useEffect, useState } from 'react';
import { useShallow } from 'zustand/react/shallow';
import { Button } from '../ui/button';
import { AddProductModal } from './AddProductModal';
import { EditProductModal } from './EditProductModal';
import { ProductSchemaType } from './ProductSchemaType';
import { useRouter } from 'next/navigation';
import { useToast } from '@frontend/hooks/use-toast';

export const MyProducts = () => {
  const { toast } = useToast();

  const {
    products,
    fetchProducts,
    createProduct,
    deleteProduct,
    updateProduct,
  } = useProductStore(useShallow((state) => ({ ...state })));

  const router = useRouter();

  const [isAddProductModalOpen, setAddProductModalOpen] = useState(false);
  const [isEditProductModalOpen, setEditProductModalOpen] = useState(false);
  const [selectedProduct, setSelectedProduct] = useState<ProductSchemaType>({
    category: '',
    description: '',
    name: '',
    price: 0,
    stockQuantity: 0,
    id: '',
    image: '',
  });

  const handleEdit = (data: ProductSchemaType) => {
    setEditProductModalOpen(false);
    setSelectedProduct(data);
  };

  const onAdd = async (data: ProductSchemaType) => {
    try {
      const fromData = new FormData();
      fromData.append('name', data.name);
      fromData.append('price', String(data.price));
      fromData.append('category', data.category);
      fromData.append('description', data.description);
      fromData.append('stockQuantity', String(data.stockQuantity));
      //fromData.append('image', data.image);
      await createProduct(fromData);
      toast({
        description: 'Product has been added successfully',
        title: 'Product added',
      });
    } catch {
      toast({
        description: 'An error occurred. Please try again',
        title: 'Error',
      });
    }
  };
  const onEdit = async (data: ProductSchemaType) => {
    try {
      const fromData = new FormData();
      fromData.append('name', data.name);
      fromData.append('price', String(data.price));
      fromData.append('category', data.category);
      fromData.append('description', data.description);
      fromData.append('stockQuantity', String(data.stockQuantity));
      //fromData.append('image', data.image);
      await updateProduct(+data.id!, fromData);
      toast({
        description: 'Product has been updated successfully',
        title: 'Product updated',
      });
    } catch {
      toast({
        description: 'An error occurred. Please try again',
        title: 'Error',
      });
    }
  };

  const onDelete = async (id: string) => {
    try {
      await deleteProduct(+id);
      toast({
        description: 'Product has been deleted successfully',
        title: 'Product deleted',
      });
    } catch {
      toast({
        description: 'An error occurred. Please try again',
        title: 'Error',
      });
    }
  };

  useEffect(() => {
    fetchProducts();
  }, [fetchProducts]);

  return (
    <section className='container mx-auto p-4 w-[900px]'>
      <h1 className='text-2xl font-bold mb-4'>My Products</h1>
      <AddProductModal
        isOpen={isAddProductModalOpen}
        onClose={(): void => {
          setAddProductModalOpen(false);
        }}
        onAdd={onAdd}
      />
      <EditProductModal
        onEdit={onEdit}
        product={selectedProduct}
        isOpen={isEditProductModalOpen}
        onClose={function (): void {
          setEditProductModalOpen(false);
        }}
      />

      <div className='flex justify-end items-center w-full'>
        <Button
          className='bg-blue-500'
          onClick={(): void => setAddProductModalOpen(true)}
        >
          <AddIcon />
          Add Product
        </Button>
      </div>

      <div className='flex justify-center items-center flex-col w-full'>
        <div className='overflow-x-auto max-w-[900px] w-full'>
          <Table className='min-w-full min-h-[900px]'>
            <TableHeader>
              <TableRow>
                <TableHead className='py-2 px-4 border-b'>Name</TableHead>
                <TableHead className='py-2 px-4 border-b'>Pice</TableHead>
                <TableHead className='py-2 px-4 border-b'>Category</TableHead>
                <TableHead className='py-2 px-4 border-b'>
                  Description
                </TableHead>
                <TableHead className='py-2 px-4 border-b'>price</TableHead>
                <TableHead className='py-2 px-4 border-b'>
                  Stock Quantity
                </TableHead>
                <TableHead className='py-2 px-4 border-b'></TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              {products.map((product) => (
                <TableRow key={product.id}>
                  <TableCell className='py-2 px-4 border-b'>
                    {product.name}
                  </TableCell>
                  <TableCell className='py-2 px-4 border-b'>
                    {product.price}
                  </TableCell>
                  <TableCell className='py-2 px-4 border-b'>
                    {product.category}
                  </TableCell>
                  <TableCell className='py-2 px-4 border-b'>
                    {product.description}
                  </TableCell>
                  <TableCell className='py-2 px-4 border-b'>
                    {product.price}
                  </TableCell>
                  <TableCell className='py-2 px-4 border-b'>
                    {product.stockQuantity}
                  </TableCell>
                  <TableCell className='py-2 px-4 border-b'>
                    <Button
                      className='mr-2 bg-blue-500'
                      onClick={() => {
                        handleEdit({
                          id: String(product.id),
                          name: product.name,
                          price: product.price,
                          category: product.category,
                          description: product.description,
                          stockQuantity: product.stockQuantity,
                          image: '',
                        });
                      }}
                    >
                      <EditIcon />
                    </Button>
                    <Button
                      className='mr-2 bg-red-500'
                      onClick={() => onDelete(String(product.id))}
                    >
                      <DeleteIcon />
                    </Button>
                    <Button
                      className='mr-2 bg-green-500'
                      onClick={() => router.push(`/product/${product.id}`)}
                    >
                      <VisibilityIcon />
                    </Button>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </div>
      </div>
    </section>
  );
};
