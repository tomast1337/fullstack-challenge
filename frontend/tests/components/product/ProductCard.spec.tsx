import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // Correct import

import { ProductDto } from '@backend/product/dto/product.dto';
import { useRouter } from 'next/router';
import { AppRouterContextProviderMock } from 'frontend/mocks/app-router-context-provider-mock';
import { ProductCard } from '@frontend/components/product/ProductCard';

// Mock the useRouter hook
jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

const mockPush = jest.fn();

(useRouter as jest.Mock).mockReturnValue({
  push: mockPush,
});

const mockProduct: ProductDto = {
  id: 1,
  name: 'Test Product',
  description: 'This is a test product',
  picture: 'https://via.placeholder.com/300',
  price: 99.99,
  stockQuantity: 10,
  createdAt: new Date(),
  updatedAt: new Date(),
};

describe('<ProductCard>', () => {
  let asFragment: () => DocumentFragment;
  let push: jest.Mock;
  beforeEach(() => {
    push = jest.fn();
    asFragment = render(
      <AppRouterContextProviderMock router={{ push }}>
        <ProductCard product={mockProduct} />
      </AppRouterContextProviderMock>,
    ).asFragment;
  });

  it('renders correctly and matches snapshot', () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the product name correctly', () => {
    expect(screen.getByText(mockProduct.name)).toBeInTheDocument();
  });

  it('should display the price formatted with two decimal places', () => {
    expect(screen.getByText('$99.99')).toBeInTheDocument();
  });

  it('should navigate to the product page when clicked', () => {
    const button = screen.getByRole('button', { name: /see product/i });
    button.click();
    expect(push).toHaveBeenCalledWith(`/product/${mockProduct.id}`);
  });

  it('should render the product image with correct src and alt attributes', () => {
    const image = screen.getByRole('img', {
      name: mockProduct.name,
    }) as HTMLImageElement;
    expect(image).toBeInTheDocument();
  });

  it("should show an 'Out of Stock' button when stockQuantity is 0", () => {
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <ProductCard product={{ ...mockProduct, stockQuantity: 0 }} />
      </AppRouterContextProviderMock>,
    );

    const outOfStockButton = screen.getByRole('button', {
      name: /out of stock/i,
    });
    expect(outOfStockButton).toBeInTheDocument();
    expect(outOfStockButton).toBeDisabled();
  });

  it("should NOT navigate when clicking 'Out of Stock' button", async () => {
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <ProductCard product={{ ...mockProduct, stockQuantity: 0 }} />
      </AppRouterContextProviderMock>,
    );

    const outOfStockButton = screen.getByRole('button', {
      name: /out of stock/i,
    });
    outOfStockButton.click();
    expect(push).not.toHaveBeenCalled();
  });
});
