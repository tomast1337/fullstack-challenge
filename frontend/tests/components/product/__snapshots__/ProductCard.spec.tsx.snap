// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard> renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border border-neutral-200 bg-white text-neutral-950 shadow-sm dark:border-neutral-800 dark:bg-neutral-950 dark:text-neutral-50 w-[300px] max-w-sm h-[500px] flex flex-col"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <img
        alt="Test Product"
        class="w-full h-48 object-cover rounded-t-lg"
        data-nimg="1"
        decoding="async"
        height="200"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F300&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F300&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F300&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
      <div
        class="text-2xl font-semibold leading-none tracking-tight mt-4"
      >
        Test Product
      </div>
      <div
        class="text-sm text-neutral-500 dark:text-neutral-400"
      >
        This is a test product
      </div>
    </div>
    <div
      class="p-6 pt-0"
    >
      <p
        class="text-lg font-semibold"
      >
        $99.99
      </p>
    </div>
    <span
      class="flex-grow"
    />
    <div
      class="flex items-center p-6 pt-0"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-white focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-neutral-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 dark:ring-offset-neutral-950 dark:focus-visible:ring-neutral-300 dark:bg-neutral-50 dark:text-neutral-900 dark:hover:bg-neutral-50/90 h-10 w-full bg-blue-500 text-white py-2 px-4 rounded-lg hover:bg-blue-600 transition-colors"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mr-2 css-1umw9bq-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
          />
        </svg>
        See Product
      </button>
    </div>
  </div>
</DocumentFragment>
`;
